sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_client_async.py

```diff
@@ -897,13 +900,53 @@ async def _message_generator_async(self, timeout=None):
             if self._shutdown:
                 await self.close_async()
 
+    async def _on_disposition_received_async(self, message_delivery, reason, state):        
+        message_delivery.reason = reason
+        if reason == LinkDeliverySettleReason.DISPOSITION_RECEIVED:
+            if state and SEND_DISPOSITION_ACCEPT in state:
+                message_delivery.state = MessageDeliveryState.Ok
+            elif state and SEND_DISPOSITION_RELEASE in state:
+                message_delivery.state = MessageDeliveryState.Ok
+            elif state and SEND_DISPOSITION_MODIFY in state:
+                message_delivery.state = MessageDeliveryState.Ok
+            elif state and SEND_DISPOSITION_RECEIVED in state:
+                message_delivery.state = MessageDeliveryState.Ok
```

Comment:
are all of these message delivery state ? If so, can we make this `if state and state in(send_disposition_accept, send_disposition_release...)`